packages/fetch-and-compile/test/fetch.test.ts

```diff
@@ -0,0 +1,130 @@
+import assert from "assert";
+import { describe, it } from "mocha";
+import Config from "@truffle/config";
+import { fetchAndCompile } from "../lib/index";
+import axios from "axios";
+import sinon from "sinon";
+const fixture: any = require("./fixture.js");
+
+describe("fetches contract on mainnet and checks for verification", () => {
+  const config = Config.default().merge({
+    networks: {
+      mainnet: {
+        url: 'https://mainnet.infura.io/v3/',
+        network_id: 1
+      }
+    },
+    network: "mainnet",
+  });
+  //stubbing axios.get method
+  beforeEach(() => {
+    //asserting that mainnet url and contract address is passed as args
+    sinon.stub(axios, 'get').withArgs("https://api.etherscan.io/api", {
+      params: {
+        module: "contract",
+        action: "getsourcecode",
+        address: '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D',
+        apikey: ''
+      },
+      responseType: "json",
+      maxRedirects: 50
+    }).returns(Promise.resolve({ data: fixture.mainnetData }))
+    //@ts-ignore
+    axios.get.callThrough();
```

Comment:
This code is duplicated twice and could be factored

Reply:
Yup yup, makes sense I refactored this to make it a bit cleaner.